### ISSUE TITLE
How to run the automation in a disconnected environment

### ISSUE THREAD

**Issue by bpradipt** [ocp-power-automation/ocp4-upi-powervm#89]
The current code pulls the following dependencies as part of execution - ocp4-helpernode, ocp4-playbooks. 
Can there be a way to download the dependencies in advance and using it ?

**Comment by bpradipt** [comment]
@yussufsh @Prajyot-Parab

**Comment by Prajyot-Parab** [comment]
@bpradipt If the bastion has a public network access the present approach won't be a issue. If not, then I believe there will be additional things to be taken care of specifically with helpernode playbook. ocp4-playbook doesn't have any hard dependencies on the public network so it should be fine in my opinion.

@arcprabh Did you try something similar on PVC? (for your last disconnected setup)

**Comment by bpradipt** [comment]
@Prajyot-Parab yeah, the assumption is bastion also doesn't have internet access (either direct or via proxy).

**Comment by yussufsh** [comment]
Full experience in isolated network is possible using an external proxy server. That is completed and tested.

On disconnected install where proxy server is not used yes these playbooks could be the problem. There can be 2 ways to solve this:
1. Download the tarballs and place it on bastion (may be a file hosted on internal HTTP server or file copied on the TF machine)
2. Run the ansible playbooks right from the TF machine. Running the playbooks from a remote machine is not tested, some tasks might not work and would need proper testing.

Assumption is that there is an enterprise registry in place with ocp images mirrored which the cluster nodes or bastion has access to.

**Comment by arcprabh** [comment]
> @bpradipt If the bastion has a public network access the present approach won't be a issue. If not, then I believe there will be additional things to be taken care of specifically with helpernode playbook. ocp4-playbook doesn't have any hard dependencies on the public network so it should be fine in my opinion.
> 
> @arcprabh Did you try something similar on PVC? (for your last disconnected setup)

@Prajyot-Parab - In my previous attempt with disconnected env, bastion had public and private ip configured and the rest of the VMs were only on private network.
